sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts

```diff
@@ -0,0 +1,547 @@
+   * @param {object}             [options]                                             The request options.
+   * @param {int}                [options.MaxSizeInMegaBytes]                          Specifies the maximum queue size in megabytes. Any attempt to enqueue a message that will cause the queue to exceed this value will fail.
+   * @param {PTnHnMnS}           [options.DefaultMessageTimeToLive]                    Depending on whether DeadLettering is enabled, a message is automatically moved to the DeadLetterQueue or deleted if it has been stored in the queue for longer than the specified time. This value is overwritten by a TTL specified on the message if and only if the message TTL is smaller than the TTL set on the queue. This value is immutable after the Queue has been created.
+   * @param {PTnHnMnS}           [options.LockDuration]                                Determines the amount of time in seconds in which a message should be locked for processing by a receiver. After this period, the message is unlocked and available for consumption by the next receiver. Settable only at queue creation time.
+   * @param {bool}               [options.RequiresSession]                             Settable only at queue creation time. If set to true, the queue will be session-aware and only SessionReceiver will be supported. Session-aware queues are not supported through REST.
+   * @param {bool}               [options.RequiresDuplicateDetection]                  Settable only at queue creation time.
+   * @param {bool}               [options.DeadLetteringOnMessageExpiration]            This field controls how the Service Bus handles a message whose TTL has expired. If it is enabled and a message expires, the Service Bus moves the message from the queue into the queue’s dead-letter sub-queue. If disabled, message will be permanently deleted from the queue. Settable only at queue creation time.
+   * @param {bool}               [options.DuplicateDetectionHistoryTimeWindow]         Specifies the time span during which the Service Bus detects message duplication.
+   * @param {bool}               [options.EnablePartitioning]                          Specifies whether the queue should be partitioned.
+   *                                                                                   `error` will contain information
+   *                                                                                   if an error occurs; otherwise `createqueueresult` will contain
+   *                                                                                   the new queue information.
+   *                                                                                   `response` will contain information related to this operation.
+   * @return {Promise<HttpOperationResponse}
+   */
+  createQueue(queuePath: string, options: any): Promise<HttpOperationResponse> {
```

Comment:
It looks like `options` should be an interface that has all the fields in the JSDoc above. Then you can document the fields on the interface and reduce the JSDoc comments.

Also since this is a TypeScript project now, you don't need to specify the types of parameters. Doc generation tools can infer those from the types used in the function signature. 

This feedback applies to all the methods where you're using `any` types in the code, but have the types documented in the JSDocs.

Reply:
I strongly agree that `options` here should be strongly typed

Reply:
Addressed all comments with code changes, please take a look @ramya-rao-a @chradek